File: 1266-minimum-time-visiting-all-points/1266-minimum-time-visiting-all-points.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minTimeToVisitAllPoints = function([[x0, y0], ...t]) {
  if (t.length <= 0) return 0
  const [[x1, y1], ..._] = t
  const dx = Math.abs(x1 - x0)
  const dy = Math.abs(y1 - y0)
  return Math.max(dx, dy) + minTimeToVisitAllPoints(t)
};

File: 1266-minimum-time-visiting-all-points/1266-minimum-time-visiting-all-points.go
func minTimeToVisitAllPoints(points [][]int) int {
  ans := 0
  for i := 1; i < len(points); i++ {
    dx, dy := points[i][0] - points[i-1][0], points[i][1] - points[i-1][1]
    if dx < 0 { dx = -dx }
    if dy < 0 { dy = -dy }
    if dx > dy { 
      ans += dx 
    } else {
      ans += dy
    }
  }
  return ans
}